src/state/session/index.tsx

```diff
@@ -368,14 +373,25 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
         persistSession: createPersistSessionHandler(
           account,
           ({expired, refreshedAccount}) => {
-            upsertAccount(refreshedAccount, expired)
+            if (expired) {
+              clearCurrentAccount()
+            } else {
+              __globalAgent = agent
+              setCurrentAccount(refreshedAccount)
```

Comment:
This persist callback is special because it swaps the `__globalClient` along with `currentAccount` at the same time.